src/components/views/rooms/RoomSettings.js

```diff
@@ -139,25 +202,91 @@ module.exports = React.createClass({
         }
         var can_set_room_avatar = current_user_level >= room_avatar_level;
 
+        var self = this;
+
+        var room_colors_section =
+            <div>
+                <h3>Room Colour</h3>
+                <div className="mx_RoomSettings_roomColors">
+                    {room_colors.map(function(room_color, i) {
+                        var selected;
+                        if (i === self.state.color_scheme_index) {
+                            selected =
+                                <div className="mx_RoomSettings_roomColor_selected">
+                                    <img src="img/tick.svg" width="17" height="14" alt="./"/>
+                                </div>
+                        }
+                        var boundClick = self.onColorSchemeChanged.bind(this, i)
```

Comment:
React complains bitterly about binding to the wrong component here, but I don't understand why.  This seems to almost precisely match the first code example at https://facebook.github.io/react/tips/expose-component-functions.html, and it's all happening inside the same component.  Bad warning?